### ISSUE TITLE
Consider adding name spacing to attributes on Checkboxes: none of the above

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-design-system#1644]
## What
Consider adding namespacing to ```data-behaviour``` to make it clearer to users where the behaviour comes from, for example ```data-govuk-exclusive=true```.
 
## Why
As part of the GOV.UK Design System working group feedback we received a recommendation to make the ```data-behaviour``` attribute consistent with other GOV.UK references and provide a more recognisable relationship to users.

## Who needs to know about this
Contributor, Vanita (Developer), Laurence (Developer)

## Anything else
[Read a background discussion on checkboxes: none of the above](https://github.com/alphagov/govuk-frontend/pull/2151).

**Comment by frankieroberto** [comment]
I like this idea, but maybe it should be consistent across other components? For example `data-govuk-module` instead of `data-module` and `data-govuk-maxlength` instead of `data-maxlength`?

(Backwards-compatibility could be maintained by continuing to support the existing un-prefixed attributes for a while?)

**Comment by 36degrees** [comment]
We previously chose not to namespace data attributes when we looked at namespacing as part of the v3.0 release.

As a result, as Frankie alluded to, none of the existing data attributes in other components are namespaced (`data-prevent-double-click`, `data-maxlength`, `data-threshold`, `data-maxwords`, `data-aria-controls` `data-disable-auto-focus`).

We should not introduce inconsistency by deciding to namespace this attribute and not the others. If we did want to change to namespacing data attributes, we should create an issue to change all of them.

Having said that, that would be a breaking change, and we want to avoid making breaking changes unnecessarily. We should only change this convention if we know that this is causing real problems. At the minute, I don't believe that's the case. 

On that basis, I'm going to close this issue. I'm not going to create another issue to change the convention at this time, but we should keep an eye out for any issues that might have been prevented by namespacing.
